business/istio_validations.go

```diff
@@ -97,11 +101,12 @@ func (in *IstioValidationsService) GetIstioObjectValidations(namespace string, o
 
 	// Get all the Istio objects from a Namespace
 	if objectType != Gateways {
-		wg.Add(3)
+		wg.Add(4)
 		// Gateways has limited fetching, others will require this information currently
 		go in.fetchDetails(&istioDetails, namespace, errChan, &wg)
 		go in.fetchServices(&services, namespace, "", errChan, &wg)
 		go in.fetchWorkloads(&workloads, namespace, errChan, &wg)
+		go in.fetchNonLocalmTLSConfigs(&mtlsDetails, errChan, &wg)
```

Comment:
Shouldn't this be for DRs only?

Reply:
Yep, DR are only needed right now. I'll remove the non-needed-yet meshpolicies.

Reply:
btw, any idea of how to fix the race condition I have in the tests?

Reply:
You do append inside each goroutine which run simultaneously. append is not thread safe, you need to use channel or equivalent.

Reply:
That is, at line 634 in ``istio_config.go``, the state of ``allDestinationRules`` is shared by each goroutine.

Reply:
https://gist.github.com/burmanm/ff447943407c9ccf5b3b79d92d6d6703

Should do the job

Reply:
Thank you! approached and tests passed.